src/core/lib/transport/metadata_batch.h

```diff
@@ -314,6 +314,20 @@ struct XEnvoyPeerMetadata : public SimpleSliceBasedMetadata {
   static absl::string_view key() { return "x-envoy-peer-metadata"; }
 };
 
+// Chaotic-good connection-type metadata trait.
```

Comment:
Let's make all of these slice-based metadata and use the string based Append() and Get() APIs on metadata_batch.

Using this form is going to add about 50 bytes per RPC for all RPCs.